websites/R/Replit/presence.ts

```diff
@@ -1,90 +1,127 @@
 const presence = new Presence({
 		clientId: "830504223153717311",
 	}),
-	browsingTimestamp = Math.floor(Date.now());
+	browsingTimestamp = Math.floor(Date.now() / 1000);
 
-presence.on("UpdateData", () => {
+let prevURL: string;
+
+const enum Assets {
+	CodeLine = "https://i.imgur.com/HNnl3oL.png",
+}
+
+presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-			details: "Viewing unsupported page",
-			largeImageKey:
-				"https://cdn.rcd.gg/PreMiD/websites/R/Replit/assets/logo.png",
+			largeImageKey: "https://i.imgur.com/rKlupJp.png",
 			startTimestamp: browsingTimestamp,
 		},
-		{ pathname } = document.location;
+		{ href, hash, pathname } = document.location,
+		buttons = await presence.getSetting<boolean>("buttons"),
+		title = document
+			.querySelector("title")
+			?.textContent?.split("- Replit")?.[0],
+		isReplit =
+			!!document.evaluate(
+				"//span[text()='Made with']",
+				document,
+				null,
+				XPathResult.FIRST_ORDERED_NODE_TYPE,
+				null
+			)?.singleNodeValue ||
+			hash.includes(".") ||
+			document
+				.querySelector('[id="__NEXT_DATA__"]')
+				?.textContent?.toLowerCase()
+				?.includes("editor");
+
+	if (!prevURL) prevURL = href;
+	else if (prevURL !== href) delete presenceData.startTimestamp;
+
+	switch (true) {
+		case pathname === "/~":
+		case pathname === "/": {
+			presenceData.details = "Viewing the homepage";
+			break;
+		}
+		case pathname.startsWith("/@"): {
+			if (isReplit) {
+				const lineNumbers = document
+						.querySelector(".cm-lineNumbers")
+						?.querySelectorAll(".cm-gutterElement"),
+					activeLine = document
+						.querySelector(".cm-lineNumbers")
+						?.querySelector(".cm-activeLineGutter")?.textContent;
+				presenceData.details = hash.split("#")?.[1]
+					? `Viewing replit: ${title} | File: ${hash.split("#")?.[1]}`
+					: `Viewing replit: ${title}`;
+				presenceData.state = `Created by: ${pathname.split("/")[1]}`;
+				presenceData.buttons = [{ label: "View Replit", url: href }];
```

Comment:
I think the more accurate term is "REPL". "Replit" is the website which contains "REPLs"